anime_downloader/sites/animesvostfr.py

```diff
@@ -0,0 +1,262 @@
+                        soup = helpers.soupify(helpers.get(urls_server[serv]))
+                    except HTTPError as err:
+                        logger.debug("_get_sources HTTP Err: %s", str(err))
+                    
+                    if soup:
+                        #logger.debug( "soup: {}".format(soup) )
+                        iframe = soup.select_one("iframe")
+                        if iframe:
+                            url = iframe['src'].strip()
+                            if url and (url > ''):
+                                logger.debug( "%s -> %s", urls_server[serv], url )
+                                urls_server[serv] = url 
+                            
+                    #video = soup.select_one("video") 
+                    #if video:
+                    #    logger.debug( "_get_sources urls: {}".format(video) )
+                        
+                    
+                    #if serv == "rapidvideo":
+                    #    sources.append(('no_extractor', urls_server[serv]))
+                    if 'youtubedownloader' in urls_server[serv]:
+                        sources.append({'extractor':'gcloud', 'server':'youtubedownloader', 'url':urls_server[serv], 'version':'subbed'})
+                        
+                    elif 'gcloud' in urls_server[serv]:
+                        sources.append({'extractor':'gcloud', 'server':'gcloud', 'url':urls_server[serv], 'version':'subbed'})
+                    
+                    elif 'feurl' in urls_server[serv]:
+                        sources.append({'extractor':'gcloud', 'server':'feurl', 'url':urls_server[serv], 'version':'subbed'})
+                            
+                    elif 'fembed' in urls_server[serv]:
+                        sources.append({'extractor':'gcloud', 'server':'fembed', 'url':urls_server[serv], 'version':'subbed'})
+                    
+                    elif 'hydrax' in urls_server[serv]:
+                        sources.append({'extractor':'hydrax', 'server':'hydrax', 'url':urls_server[serv], 'version':'subbed'})
+                        
+                    else:
+                        logger.warn( "_get_sources url: %s not supported", urls_server[serv])
+               
+            # TODO: put it into config.py   
+            if not self.sitename in Config['siteconfig']:         
+                Config['siteconfig'][self.sitename] = {
+                    "servers": [
+                        "youtubedownloader",
+                        "gcloud",
+                        "feurl",
+                        "fembed",
+                        "hydrax"
+                    ]
+                } 
+            
+            # TODO: Add These proterties to base_extractor
+            extGcloud = get_extractor('gcloud')
+            if extGcloud:
+                setattr(extGcloud, 'extractorVersion', 2)
+                setattr(extGcloud, 'fallback_qualities', self._parent._fallback_qualities)
```

Comment:
Don't use qualities this way, only use self.quality or convince vn-ki otherwise, the extractors will get re-run with changed self.quality (based on fallback_qualities) for every iteration, feel free to join the discord for better discussion/feedback

Reply:
I think the best way is:
- Add property extractorVersion into Anime
- Add properties extractorVersion and fallback_qualities and change the constructor into base_extractor
- Modify extractor like my modifcations into gcloud by test extractorVersion (by defaut the last way is V1)

Reply:
@vn-ki take a look

Reply:
maybe in the config, i could modify like this (add extractorVersion) ? :

```
"siteconfig": {
    'animesvostfr': {
            "servers": [
                "youtubedownloader",
                "gcloud",
                "feurl",
                "fembed",
                "hydrax"
            ]
            "extractorVersion": {
                "hydrax": 2
                "gcloud": 2
            }
        },
```

Reply:
Modifying extractor behaviour in provider is unacceptable

Reply:
it was temporary, it was not of actuality